pkg/client/client.go

```diff
@@ -829,6 +829,12 @@ func (c *KpmClient) AddDepToPkg(kclPkg *pkg.KclPkg, d *pkg.Dependency) error {
 		return err
 	}
 
+	dep, err := c.LoadPkgFromPath(c.getDepStorePath(c.homePath, d, false))
+	if err != nil {
+		return fmt.Errorf("failed to load downloaded package: %w", err)
+	}
+	d.Source.Git.Version = dep.GetPkgVersion()
```

Comment:
Here might be some other cases, not just git, if the dependency is not from git repo, `d.Source.Git` will be `nil`.